pandas/core/groupby/grouper.py

```diff
@@ -558,9 +557,12 @@ def __init__(
                 raise AssertionError(errmsg)
 
         if isinstance(self.grouping_vector, np.ndarray):
-            # if we have a date/time-like grouper, make sure that we have
-            # Timestamps like
-            self.grouping_vector = sanitize_to_nanoseconds(self.grouping_vector)
+            if self.grouping_vector.dtype.kind in ["m", "M"]:
+                # if we have a date/time-like grouper, make sure that we have
+                # Timestamps like
+                # TODO 2022-10-08 we only have one test that gets here and
+                #  values are already in nanoseconds in that case.
+                self.grouping_vector = Series(self.grouping_vector).to_numpy()
```

Comment:
So `Series(...).to_numpy()` will coerce to nanos and then output to numpy? Is this a lot slower than `sanitize_to_nanoseconds`?

Reply:
> So Series(...).to_numpy() will coerce to nanos and then output to numpy?

Yes.

> Is this a lot slower than sanitize_to_nanoseconds?

A bit.  Adds about 65 microseconds.